### ISSUE TITLE
WhatsApp Gupshup Integration

### ISSUE THREAD

**Issue by converso-chat** [chatwoot/chatwoot#1119]
** Is your feature request related to a problem? Please describe. **
This is not a problem. I would like to suggest the integration of Gupshup WhatsApp. The rates are much more attractive than Twilio, in addition to having interactive buttons in the message templates.

** Describe the solution you'd like **
Integrate gupshup.io

** Describe alternatives you've considered **
Gupshup API is much better to work with, in addition to what I already mentioned we can use our own numbers to approve with WhatsApp and not rent as in Twilio

** Additional context **
I hope you think about the positive points for this integration

**Comment by iammattmartin** [comment]
The better solution would be native integration with WhatsApp.

Otherwise, you'll see lots of "please integrate with X to get Y".

**Comment by pranavrajs** [comment]
@iammattmartin Some businesses might not have the Whatsapp Business API access, it is advised to go through a vendor to get the access. This is one of the reasons why didn't go with the actual Whatsapp API implementation rather selected a vendor that works for most of the countries.

**Comment by iammattmartin** [comment]
> @iammattmartin Some businesses might not have the Whatsapp Business API access

I agree. My comment was more inferring that CW may opt to integrate direct and become the gateway than have to integrate with every possible aggregator like Twilio or Gupshub etc.

**Comment by converso-chat** [comment]
Gupshup is like Twilio. However, we can use our own numbers, it also has options of Interactive buttons with Call to Action, Quick Reply, call ... Besides costing only $ 0.001 and Twilio $ 0.005. Gupshup is much better than Twilio. But it is just a suggestion of mine.

**Comment by galaxy79** [comment]
@pranavrajs is there a plan to integrate Gupshup in future?

**Comment by converso-chat** [comment]
Hi! I would agree to hire someone to help with this implementation. Gupshup is excellent. If anyone else is interested we can share and fund this implementation

**Comment by iammattmartin** [comment]
They can do work, for a fee:

https://www.chatwoot.com/pricing

You'd need to consider their enterprise pricing though for custom work if you cannot do it yourself or find someone to do it for you.

**Comment by mariusa** [comment]
> Some businesses might not have the Whatsapp Business API access, it is advised to go through a vendor to get the access. This is one of the reasons why didn't go with the actual Whatsapp API implementation

For businesses which do have Whatsapp Business API access, it would still be useful to have a native integration vs. having to setup & pay a 3rd party provider.

**Comment by thundersparkf** [comment]
I was experimenting with the Gupshup integration and I am facing this issue. 

<img width="292" alt="Screenshot 2021-06-26 at 12 47 59 AM" src="https://user-images.githubusercontent.com/64605024/123474815-28fa8d00-d618-11eb-93fc-5ffce0a3565a.png">

Seems to be similar to the Telegram and Line integrations, where exactly do I look to clear this up?

**Comment by iammattmartin** [comment]
I can't see any commits for Gupshup, have you coded this yourself?

**Comment by thundersparkf** [comment]
Yeah. I'm new to Ruby on Rails, so I didn't want to push it yet. Sorry if the error description is too vague, I'm just trying to understand why this happens, cause I'm assuming Telegram and Line integrations would have a similar reason, so I can vague direction to move in.

**Comment by iammattmartin** [comment]
I'd recommend asking in the discussion tab or opening a new issue, as this one is for the FR. You could always link this back as PR once you've sorted that.

**Comment by thundersparkf** [comment]
Cool makes sense. Thank you!

**Comment by iammattmartin** [comment]
Would be nice to see if this can be added. Twilio's verification is fairly painful, to the point we've stopped using them.
